extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -15,43 +15,115 @@
  */
 
 import {isExperimentOn} from '../../../src/experiments';
-import {getService} from '../../../src/service';
-import {dev} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+import {viewerFor} from '../../../src/viewer';
+import {Layout} from '../../../src/layout';
+import {dev, user} from '../../../src/log';
 
 /** @private @const {string} */
 const TAG = 'amp-share-tracking';
 
-class AmpShareTracking extends AMP.BaseElement {
+export class AmpShareTracking extends AMP.BaseElement {
+  /**
+    * @return {boolean}
+    * @private
+    */
+  isExperimentOn_() {
+    return isExperimentOn(this.win, TAG);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
```

Comment:
I created a separate [PR](https://github.com/ampproject/amphtml/pull/4234) since this has nothing to do with share-tracking.